sec3/neovim.md

```diff
@@ -0,0 +1,207 @@
+
+Here are the commands to create and delete tabs:
+
+ * `:tabnew` or `:tabe` - Open a new tab.
+ * `:tabclose` or `:tabc` - Close the current tab.
+ * `:tabonly` or :`tabo` - Close every other tab except the current one.
+
+To move from tab to tab, you can use these keystrokes:
+
+ * `gt` - go to the next tab.
+ * `gT` - go to the previous tab.
+
+You can also add a count before the last two keystrokes. For example, 1gT goes to the first tab. Worth remembering that tabs are indexed from 1, not from 0.
+
+### Argument List (arglist)
+
+The argument list (also called arglist) is the fourth and last abstraction allowing you to organize your open files. It’s useful to see it as a `stable subset` of the buffer list. When we ran `vim file1 file2 file3 file4`
+
+ 1. Every file in the arguments above will be in the buffer list and the arglist
+ 2. Some files in the buffer list might not be in the arglist
+
+The arglist can be useful to isolate some files from the buffer list to do some operations on them. Here are some commands you can use to manipulate the arglist:
+
+  * `:args` - Display the arglist.
+  * `:argadd` - Add file to the arglist.
+  * `:argdo` - Execute a command on every file in the arglist.
+
+To edit the files in the arglist, you can use these commands:
+
+  * `:next` - Move to the next file in the arglist.
+  * `:prev` - Move to the previous file in the arglist.
+  * `:first` - Move to the first file in the arglist.
+
+### Mapping Keystrokes
+
+Mapping keystrokes eliminates toil from repetition. Create a file called `file_extraction.txt` with the following content: 
+
+
+```txt
+The quick brown fox jumps over the lazy dog
+The quick brown fox jumps over the lazy dog
+The quick brown fox jumps over the lazy dog
+The quick brown fox jumps over the lazy dog
+The quick brown fox jumps over the lazy dog
+The quick brown fox jumps over the lazy dog
+```
+
+To work with this files we will demonstrate a simple example of remapped keys that mean less typing. For that one uses commands for all the Neovim modes:
+
+  * `:nnoremap` - Create a new mapping for NORMAL mode (non recursive).
+  * `:inoremap` - Create a new mapping for INSERT mode (non recursive).
+  * `:vnoremap` - Create a new mapping for VISUAL mode (non recursive).
+
+As an example let's map `s` to `dd` Keep in mind `s`ubstitute is already used in your default mapping of a Neovim installation. Use the following commands on the file you just created.
+
+  1. Run the `ex` command `:noremap s dd`
+  2. Try to hit the keystroke `dd`. It will delete a line. (hit `u` to undo the delete)
+  3. Try to hit `s`. It deletes a line too.
+
+If you close your Neovim this silly change will vanish. The lesson here is avoid using the default Neovim's mappings. 
```

Comment:
Avoid using the default mappings? In general it is good to use and know defaults so that you can function on systems outside your own home machine. I would advise to enhance the default mappings by adding additional mappings as shortcuts for commonly used functionality.

Reply:
I meant the opposite.